opensearch_upload/workflow_metrics_uploader.py

```
@@ -0,0 +1,795 @@
+            print(f"❌ Failed to fetch jobs for run {run_id}: {e}")
+            return None
+    
+    def calculate_time_diff(self, start_time: str, end_time: str) -> int:
+        """Calculate duration in seconds"""
+        if not start_time or not end_time:
+            return 0
+        try:
+            start_dt = datetime.fromisoformat(start_time.replace('Z', '+00:00'))
+            end_dt = datetime.fromisoformat(end_time.replace('Z', '+00:00'))
+            return max(0, int((end_dt - start_dt).total_seconds()))
+        except:
```

Ruff (0.14.14)

[error] Do not use bare `except`

(E722)